api/v1beta2/awsmachine_types.go

```diff
@@ -197,6 +197,14 @@ type AWSMachineSpec struct {
 	// CapacityReservationID specifies the target Capacity Reservation into which the instance should be launched.
 	// +optional
 	CapacityReservationID *string `json:"capacityReservationId,omitempty"`
+
+	// MarketType specifies the type of market for the EC2 instance. Valid values include:
+	// "spot": The instance runs as a Spot instance. When SpotMarketOptions is provided, the MarketType defaults to "spot".
+	// "capacity-block": The instance utilizes pre-purchased compute capacity (capacity blocks) with AWS Capacity Reservations.
+	//  If this value is selected, CapacityReservationID must be specified to identify the target reservation.
+	// If MarketType is not specified and SpotMarketOptions is provided, the MarketType defaults to "spot".
+	// +optional
+	MarketType *MarketType `json:"marketType,omitempty"`
```

Comment:
Does this need to be a pointer?

Reply:
It should allow empty values so I thought we should use a pointer here.

Reply:
Do you care about the difference between `marketType: ""` and the field not being present at all? The controller will treat those two representations in the same way won't it?

Reply:
Hmm, I see then I'll make a change. Thanks